From a2a4471ca21b04419102138d1d062eb380cb9424 Mon Sep 17 00:00:00 2001
From: commandblock2 <tony_hu@live.com>
Date: Fri, 14 Feb 2025 02:21:16 +0800
Subject: [PATCH] feat-script-registration-info

---
 .../script/PolyglotScript$RegisterScript.d.ts        | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript$RegisterScript.d.ts b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript$RegisterScript.d.ts
index c6a30ed..f1ed2a6 100644
--- a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript$RegisterScript.d.ts
+++ b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript$RegisterScript.d.ts
@@ -1,7 +1,15 @@
 import type { Object } from '../../../../java/lang/Object.d.ts'
 import type { Function } from '../../../../java/util/function/Function.d.ts'
 import type { PolyglotScript } from '../../../../net/ccbluex/liquidbounce/script/PolyglotScript.d.ts'
-export class PolyglotScript$RegisterScript extends Object implements Function<{ [key: string]: Object }, PolyglotScript> {
+
+type ScriptInfo = {
+    name: string;
+    version: string;
+    authors: string[];
+};
+
+export class PolyglotScript$RegisterScript extends Object implements Function<ScriptInfo, PolyglotScript> {
     constructor(null: PolyglotScript$RegisterScript)
-    apply(scriptObject: { [key: string]: Object }): PolyglotScript;
+
+    apply(scriptObject: ScriptInfo): PolyglotScript;
 }
\ No newline at end of file
-- 
2.47.1

