From 9a078fecb8814d0fcd16856257bae9e4ab13ccdd Mon Sep 17 00:00:00 2001
From: commandblock2 <tony_hu@live.com>
Date: Fri, 14 Feb 2025 00:24:31 +0800
Subject: [PATCH] feat-module-registration-info

---
 .../ccbluex/liquidbounce/script/PolyglotScript.d.ts | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript.d.ts b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript.d.ts
index e0a9f39..e70db91 100644
--- a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript.d.ts
+++ b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/script/PolyglotScript.d.ts
@@ -10,6 +10,17 @@ import type { ClientModule } from '../../../../net/ccbluex/liquidbounce/features
 import type { ChoiceConfigurable } from '../../../../net/ccbluex/liquidbounce/config/types/ChoiceConfigurable.d.ts'
 import type { Function1 } from '../../../../kotlin/jvm/functions/Function1.d.ts'
 import type { Value } from '../../../../org/graalvm/polyglot/Value.d.ts'
+import type { ScriptModule } from './bindings/features/ScriptModule.d.ts'
+
+interface ModuleObject {
+    name: string;
+    category: string;
+    description: string;
+    settings?: {
+        [key: string]: Value
+    };
+}
+
 export class PolyglotScript extends Object {
     constructor(language: string, file: File, debugOptions: ScriptDebugOptions)
     // private getContext(): Context;
@@ -33,5 +44,5 @@ export class PolyglotScript extends Object {
     on(eventName: string, handler: Function0<Unit>): Unit;
     registerChoice(choiceConfigurable: ChoiceConfigurable<Choice>, choiceObject: { [key: string]: Object }, callback: Function1<Choice, Unit>): Unit;
     registerCommand(commandObject: Value): Unit;
-    registerModule(moduleObject: { [key: string]: Object }, callback: Function1<ClientModule, Unit>): Unit;
+    registerModule(moduleObject: ModuleObject, callback: (module: ScriptModule) => void): Unit;
 }
\ No newline at end of file
-- 
2.47.1

