From ddebf1433abbe922f1b984157036a239582fc078 Mon Sep 17 00:00:00 2001
From: commandblock2 <tony_hu@live.com>
Date: Fri, 14 Feb 2025 01:03:37 +0800
Subject: [PATCH] fix-script-value-get-value

---
 .../liquidbounce/config/types/Value.d.ts      | 31 +++++++++++++------
 1 file changed, 21 insertions(+), 10 deletions(-)

diff --git a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/config/types/Value.d.ts b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/config/types/Value.d.ts
index 669a829..f96ee3e 100644
--- a/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/config/types/Value.d.ts
+++ b/minecraft-yarn-definitions/types/net/ccbluex/liquidbounce/config/types/Value.d.ts
@@ -34,7 +34,7 @@ export class Value<T extends Object> extends Object {
     doNotIncludeWhen(condition: Function0<boolean>): Value<T>;
     get(): T;
     // getValue(u: Object | null, property: KProperty<Object | null>): T;
-    getValue(): T;
+    // getValue(): T;
     independentDescription(): Value<T>;
     notAnOption(): Value<T>;
     onChange(listener: Function1<T, T>): Value<T>;
@@ -45,18 +45,29 @@ export class Value<T extends Object> extends Object {
     setByString(string: string): Unit;
     // setValue(u: Object | null, property: KProperty<Object | null>, t: T): Unit;
     // setValue(t: Value): Result<Unit>;
-    setValue(value: T): Unit;
+    // setValue(value: T): Unit;
     type(): ValueType;
 
-    getValue(): T extends ClosedFloatingPointRange<number> ? number[] : T;
-    setValue(value: T extends ClosedFloatingPointRange<number> ? number[] | T : T): Unit;
+    // getValue(): T extends ClosedFloatingPointRange<number> ? number[] : T;
+    // setValue(value: T extends ClosedFloatingPointRange<number> ? number[] | T : T): Unit;
 
-    // Special case for key values
-    getValue(): T extends InputUtil$Key ? string : T;
-    setValue(value: T extends InputUtil$Key ? string | T : T): Unit;
+    // // Special case for key values
+    // getValue(): T extends InputUtil$Key ? string : T;
+    // setValue(value: T extends InputUtil$Key ? string | T : T): Unit;
 
-    // Special case for named choices
-    getValue(): T extends NamedChoice ? string : T;
-    setValue(value: T extends NamedChoice ? string | T : T): Unit;
+    // // Special case for named choices
+    // getValue(): T extends NamedChoice ? string : T;
+    // setValue(value: T extends NamedChoice ? string | T : T): Unit;
+
+
+    getValue(): T extends ClosedFloatingPointRange<number> ? number[] :
+        T extends InputUtil$Key ? string :
+        T extends NamedChoice ? string :
+        T;
+
+    setValue(value: T |
+        (T extends ClosedFloatingPointRange<number> ? number[] : never) |
+        (T extends InputUtil$Key ? string : never) |
+        (T extends NamedChoice ? string : never)): Unit;
 
 }
\ No newline at end of file
-- 
2.47.1

